### ISSUE TITLE
Deprecated note in set documentation

### ISSUE THREAD

**Issue by JulienPalard** [python/cpython#74684]
BPO | [30499](https://bugs.python.org/issue30499)
--- | :---
Nosy | @rhettinger, @JulienPalard, @Mariatta
PRs | <li>python/cpython#1848</li><li>python/cpython#1905</li><li>python/cpython#1906</li><li>python/cpython#1907</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/Mariatta'
closed_at = <Date 2017-06-01.14:13:51.398>
created_at = <Date 2017-05-28.21:30:17.928>
labels = ['3.7', 'docs']
title = 'Deprecated note in set documentation'
updated_at = <Date 2017-06-01.14:13:51.396>
user = 'https://github.com/JulienPalard'
```

bugs.python.org fields:
```python
activity = <Date 2017-06-01.14:13:51.396>
actor = 'Mariatta'
assignee = 'Mariatta'
closed = True
closed_date = <Date 2017-06-01.14:13:51.398>
closer = 'Mariatta'
components = ['Documentation']
creation = <Date 2017-05-28.21:30:17.928>
creator = 'mdk'
dependencies = []
files = []
hgrepos = []
issue_num = 30499
keywords = []
message_count = 7.0
messages = ['294662', '294904', '294940', '294941', '294942', '294943', '294944']
nosy_count = 4.0
nosy_names = ['rhettinger', 'docs@python', 'mdk', 'Mariatta']
pr_nums = ['1848', '1905', '1906', '1907']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue30499'
versions = ['Python 2.7', 'Python 3.5', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by JulienPalard** [comment]
While translating doc to french, I found a surprising comment in stdtypes.html about sets, stating:

> To support searching for an equivalent frozenset, the elem set is temporarily mutated during the search and then restored. During the search, the elem set should not be read or mutated since it does not have a meaningful value.

So I searched the equivalent in the code, at least to help me understand it to translate it in a proper way, but found it has been deleted in 51ced7afe72d861b46f069d757f2787f13253d32.

This commit is in branches 2.7 and 3.1 to 3.6 (and master).

**Comment by rhettinger** [comment]
This is good to go.

**Comment by Mariatta** [comment]
New changeset 0737ee20671d60802a7ef40f653fe449f1921de0 by Mariatta (Julien) in branch 'master':
bpo-30499: Remove a deprecated note about sets. (GH-1848)
https://github.com/python/cpython/commit/0737ee20671d60802a7ef40f653fe449f1921de0

**Comment by Mariatta** [comment]
New changeset 9d752aa5e695d682f3cfc40655580900b132e52a by Mariatta in branch '3.6':
bpo-30499: Remove a deprecated note about sets. (GH-1848) (GH-1905)
https://github.com/python/cpython/commit/9d752aa5e695d682f3cfc40655580900b132e52a

**Comment by Mariatta** [comment]
New changeset 3419fc2bf1e81d42ef4578e59d3c10f9c239e80f by Mariatta in branch '3.5':
bpo-30499: Remove a deprecated note about sets. (GH-1848) (GH-1906)
https://github.com/python/cpython/commit/3419fc2bf1e81d42ef4578e59d3c10f9c239e80f

**Comment by Mariatta** [comment]
New changeset 662d856cd7d21dbdbd4043c1735d93b32568c9e5 by Mariatta in branch '2.7':
bpo-30499: Remove a deprecated note about sets. (GH-1848) (GH-1907)
https://github.com/python/cpython/commit/662d856cd7d21dbdbd4043c1735d93b32568c9e5

**Comment by Mariatta** [comment]
Thanks Julien and Raymond.
PR has been merged and backported to 2.7, 3.5 and 3.6.
